extensions/data-workspace/src/common/interfaces.ts

```diff
@@ -97,4 +97,18 @@ export interface IWorkspaceService {
 	 * Shows confirmation message that the extension host will be restarted and current workspace/file will be closed. If confirmed, the specified workspace will be entered.
 	 */
 	enterWorkspace(workspaceFile: vscode.Uri): Promise<void>;
+
+	/**
+	 * Gets the project actions to be placed on the dashboard toolbar
+	 * @param provider The project provider
+	 */
+	getProjectToolbarActions(provider: IProjectProvider): IProjectAction[];
+
+	/**
+	 * Perform the desired action from the tooldbar
+	 * @param provider The project provider
+	 * @param treeNode The treeItem in a project's hierarchy, to be used to obtain a Project
+	 * @param actionId the action to perform
+	 */
+	performAction(provider: IProjectProvider, treeItem: WorkspaceTreeItem, actionId: string): Promise<void>;
```

Comment:
I don't think we need this method, you can add a run(context) method to the action.

Reply:
Fixed it in the latest iteration. Thanks!